language/testing-infra/diem-transactional-test-runner/src/diem_test_harness.rs

```diff
@@ -0,0 +1,320 @@
+use anyhow::Result;
+use diem_crypto::ed25519::{Ed25519PrivateKey, Ed25519PublicKey};
+use diem_crypto::ValidCryptoMaterialStringExt;
+use diem_types::account_config::{self, XUS_NAME};
+use diem_types::{
+    chain_id::ChainId,
+    transaction::{
+        Module as TransactionModule, RawTransaction, Script as TransactionScript,
+        ScriptFunction as TransactionScriptFunction,
+    },
+};
+use language_e2e_tests::executor::FakeExecutor;
```

Comment:
Do we need to pull this in? I was imagining it would be easy enough to just talk to the diem adapter directly

Reply:
Looking at it a bit closer, it should be easier now than it was in the past to just talk to diem_vm.rs right?

Reply:
Agreed we should get rid of it, but I'm currently depending on it to fetch some on-chain resources. Let me see what I can do. I'll try to get rid of it, but I have feeling that this will be a lot of work. If that's the case then I'll likely do it in a separate PR.

Reply:
I'm thinking it might actually be easier to start with a very basic invocation of `DiemVM` + `InMemoryStorage`. Reading or editing these stateful resources probably isn't needed in the first PR

Reply:
> Reading or editing these stateful resources probably isn't needed in the first PR

It's needed because we need to fetch certain transaction parameters which are again, required to perform any transactions. If you don't fetch the default values from the chain, you'll have to require many more additional command arguments, which will be quite messy IMO.

Reply:
Ah gotcha

... I would still be curious to see one without pulling in FakeExecutor though :P Our goal is to kill language/e2e-testsuite as its all kind of crufty and out dated. My unsubstantiated and unresearched belief is that the new thing can be built a lot simpler